src/components/ToggleButton.js

```diff
@@ -0,0 +1,36 @@
+import React, {PropTypes} from 'react'
+import styles from './ToggleButton.css'
+import CSSModules from 'react-css-modules'
+import {changeTokenType} from '../actions/actions.js'
+import getID from '../utils/getID'
+
+class ToggleButton extends React.Component {
+  componentWillMount () {
+    this.state = {isChecked: false}
+    this.onChange = this.onChange.bind(this)
+    this.toggleId = getID()
+  }
+  onChange (e) {
+    let nextVal = !this.state.isChecked
+    this.setState({
+      isChecked: nextVal
+    })
+    // we need to inverse the nextVal because checked means isPreview === false
```

Comment:
- does it mean that the previous version was wrong? (you used `!state.isChecked` thrice)
- better name for this variable: `toggledValue` (indicates that it's previous value, but toggled)
- you shouldn't mention `isPreview` here as this is just a reusable component that can be used in any context